tcf_website/tests/test_review.py

```diff
@@ -0,0 +1,57 @@
+# pylint: disable=no-member
+"""Tests for Review model."""
+
+from django.contrib.messages import get_messages
+from django.test import TestCase
+from django.urls import reverse
+
+
+from .test_utils import setup
+
+
+class DeleteReviewTests(TestCase):
+    """Tests for the DeleteReview view."""
+
+    def setUp(self):
+        setup(self)  # set up tests: add some example data
+
+    def test_delete_review_message(self):
+        """Test if a message is shown when a user deletes their review."""
+        self.client.force_login(
+            self.review1.user)  # force a login as the author of review1
+        # try and make the user delete review1
+        response = self.client.post(
+            reverse('delete_review', args=[self.review1.id]),
+        )
+
+        self.assertEqual(response.status_code, 302)  # ensure 302 Found status
+
+        # get messages from the request
+        messages = [m.message for m in get_messages(response.wsgi_request)]
+        self.assertEqual(str(messages[0]), 'Successfully deleted your review!')
+
+    def test_delete_nonexistent_review_id(self):
+        """Test if a 404 is returned for deleting a nonexistent review ID."""
+        self.client.force_login(self.user1)
+        response = self.client.post(
+            reverse('delete_review', args=[0])  # id 0 = nonexistent review
+        )
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_unauthorized_user_delete(self):
+        """Test if a 403 is returned for unauthorized review deletion."""
+        self.client.force_login(self.user2)  # force login as user2
+
+        # Note: because PermissionDenied is (should be) raised in this view, we get a really ugly
+        # traceback in the test runner. This isn't so nice, because it makes the
+        # `./precommit` output annoying to sort through.
```

Comment:
bless you for actually remembering to use `precommit` lmao

Reply:
On a more serious note, I spent a bit of time trying to figure out a way to solve the problem, but the best I got was setting `sys.tracebacklimit=0`, which gets rid of most of the stack trace, yielding this:

![image](https://user-images.githubusercontent.com/55100084/105948269-aa81cb00-6038-11eb-915f-8a7519ea7bba.png)

Also, saw some [discussion on StackOverflow](https://stackoverflow.com/questions/8391411/how-to-block-calls-to-print) about blocking print statements entirely by redirecting `sys.stdout`, but when I tried it, the errors still printed out all the same.

The reduction is still an improvement, but not enough IMO. This is definitely something we want to figure out before merging into `dev`.